apis/v1alpha2/httproute_types.go

```diff
@@ -70,20 +70,33 @@ type HTTPRouteSpec struct {
 	// HTTPRoute rules. If no hostname is specified, traffic is routed
 	// based on the HTTPRouteRules.
 	//
-	// Hostname can be "precise" which is a domain name without the terminating
-	// dot of a network host (e.g. "foo.example.com") or "wildcard", which is
-	// a domain name prefixed with a single wildcard label (e.g. `*.example.com`).
-	// The wildcard character `*` must appear by itself as the first DNS
-	// label and matches only a single label.
-	// You cannot have a wildcard label by itself (e.g. Host == `*`).
 	// Requests will be matched against the Host field in the following order:
 	//
-	// 1. If Host is precise, the request matches this rule if
-	//    the HTTP Host header is equal to Host.
-	// 2. If Host is a wildcard, then the request matches this rule if
+	// 1. If Hostname is precise, the request matches this rule if
+	//    the HTTP Host header is equal to the Hostname.
+	// 2. If Hostname is a wildcard, then the request matches this rule if
 	//    the HTTP Host header is to equal to the suffix
 	//    (removing the first label) of the wildcard rule.
 	//
+	// If a hostname is specified by the Listener that the HTTPRoute is bound
+	// to, at least one hostname specified here must match the Listener specified
```

Comment:
I think there's still a bit of nuance here. The absence of hostnames on Routes or Listeners is interpreted as "*", so this is only the case if hostnames are specified on both the Route and the Listener.

Reply:
Updated again. Hopefully this should make it all explicit. Thanks for careful reading, I think we're getting closer to something that's properly unambiguous.